File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let release = [];
    let left_day = [];
    
    for(let i = 0; i < progresses.length; i++) {
        left_day.push(Math.ceil((100 - progresses[i]) / speeds[i]));
    }
    
    
    let day = 1;
    let left = left_day.shift();
    while(left_day.length !== 0) {
        if(left >= left_day[0]) {
            day++;
            left_day.shift();
        } else {
            release.push(day);
            left = left_day.shift();
            day = 1;
        }
    }
    release.push(day);

    return release;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque


def solution(progresses, speeds):
    q = deque()
    s = deque()
    answer = []
    
    q.extend(progresses)
    s.extend(speeds)
    
    while q:
        while q[0] < 100:
            for i in range(len(s)):
                q[i] += s[i]
        
        print(q)
        cnt = 0
        while q and q[0] >= 100:
            cnt += 1
            q.popleft()
            s.popleft()
        answer.append(cnt)
    return answer